src/apple/macos-system-helper.ts

```diff
@@ -0,0 +1,64 @@
+import { execSync } from 'child_process';
+import { debug } from '../utils/debug';
+
+export class MacOSSystemHelpers {
+  static findSDKRootDirectoryPath(): string | undefined {
+    try {
+      const sdkPath = execSync('xcrun --show-sdk-path', {
+        encoding: 'utf8',
+      }).trim();
```

Comment:
In CI this resolves to the Command Line Tools:

```
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
```

Locally it resolves to the Xcode.app:

```
/Applications/Xcode-16.3.0.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk
```